tools/nightly-builder/build-required.js

```diff
@@ -0,0 +1,49 @@
+"use strict"
+
+const strftime     = require('strftime').timezone(0)
+    , after        = require('after')
+    , latestBuild  = require('./latest-build')
+    , latestCommit = require('./latest-commit')
+
+const dateFormat = '%Y%m%d'
+
+
+function timeString () {
+  return strftime(dateFormat)
+}
+
+
+function buildRequired (type, callback) {
+  let done = after(2, onData)
+    , build
+    , commit
+
+  function onData (err) {
+    if (err)
+      return callback(err)
+
+    let buildCommit = build.commit.substr(0, 10)
+      , buildDate   = strftime(dateFormat, build.date)
```

Comment:
Rright, so this is something I wanted sanity check on - I originally made it compare commit _and_ date but then rolled back to just commit check. But this means two things can happen:

1. if the program is run manually and the head commit has changed it'll just make a new nightly, this is not a big problem because I don't expect it to be run manually but it could be useful
2. if you manually create a nightly earlier in the day and the head commit has moved on, this program is going to trigger another nightly build, and perhaps that's undesirable.

We could enforce one-nightly-per-day with this but at the moment it's not.